src/all_types.hpp

```diff
@@ -2527,20 +2534,31 @@ struct IrInstructionBinOp {
     bool safety_check_on;
 };
 
-struct IrInstructionLoadPtr {
+struct IrInstructionLoad {
     IrInstruction base;
 
     IrInstruction *ptr;
 };
 
+struct IrInstructionExtractInsert {
+    IrInstruction base;
+
+    IrInstruction *agg;
+    IrInstruction *index;
+    // nullptr if Extract
```

Comment:
Looks like this doesn't need to be separate fields?

Reply:
No, these extract fields, and all the others, are used for Extract

Reply:
No, these extract fields, and all the others, are used for Extract.